src/eval.c

```diff
@@ -352,217 +352,6 @@ do_debug_on_call (Lisp_Object code, ptrdiff_t count)
-      val = eval_sub (form);
-    }
-
-  return val;
-}
-
-/* Evaluate BODY sequentially, discarding its value.  */
-
-void
-prog_ignore (Lisp_Object body)
-{
-  Fprogn (body);
-}
-
-DEFUN ("prog1", Fprog1, Sprog1, 1, UNEVALLED, 0,
-       doc: /* Eval FIRST and BODY sequentially; return value from FIRST.
-The value of FIRST is saved during the evaluation of the remaining args,
-whose values are discarded.
-usage: (prog1 FIRST BODY...)  */)
-  (Lisp_Object args)
-{
-  Lisp_Object val = eval_sub (XCAR (args));
-  prog_ignore (XCDR (args));
-  return val;
-}
-
-DEFUN ("prog2", Fprog2, Sprog2, 2, UNEVALLED, 0,
-       doc: /* Eval FORM1, FORM2 and BODY sequentially; return value from FORM2.
-The value of FORM2 is saved during the evaluation of the
-remaining args, whose values are discarded.
-usage: (prog2 FORM1 FORM2 BODY...)  */)
-  (Lisp_Object args)
-{
-  eval_sub (XCAR (args));
-  return Fprog1 (XCDR (args));
-}
-
-DEFUN ("setq", Fsetq, Ssetq, 0, UNEVALLED, 0,
-       doc: /* Set each SYM to the value of its VAL.
-The symbols SYM are variables; they are literal (not evaluated).
-The values VAL are expressions; they are evaluated.
-Thus, (setq x (1+ y)) sets `x' to the value of `(1+ y)'.
-The second VAL is not computed until after the first SYM is set, and so on;
-each VAL can use the new value of variables set earlier in the `setq'.
-The return value of the `setq' form is the value of the last VAL.
-usage: (setq [SYM VAL]...)  */)
-  (Lisp_Object args)
-{
-  Lisp_Object val = args, tail = args;
-
-  for (EMACS_INT nargs = 0; CONSP (tail); nargs += 2)
-    {
-      Lisp_Object sym = XCAR (tail), lex_binding;
-      tail = XCDR (tail);
-      if (!CONSP (tail))
-	xsignal2 (Qwrong_number_of_arguments, Qsetq, make_number (nargs + 1));
-      Lisp_Object arg = XCAR (tail);
-      tail = XCDR (tail);
-      val = eval_sub (arg);
-      /* Like for eval_sub, we do not check declared_special here since
-	 it's been done when let-binding.  */
-      if (!NILP (Vinternal_interpreter_environment) /* Mere optimization!  */
-	  && SYMBOLP (sym)
-	  && !NILP (lex_binding
-		    = Fassq (sym, Vinternal_interpreter_environment)))
-	XSETCDR (lex_binding, val); /* SYM is lexically bound.  */
-      else
-	Fset (sym, val);	/* SYM is dynamically bound.  */
-    }
-
-  return val;
-}
-
-DEFUN ("function", Ffunction, Sfunction, 1, UNEVALLED, 0,
-       doc: /* Like `quote', but preferred for objects which are functions.
-In byte compilation, `function' causes its argument to be compiled.
-`quote' cannot do that.
-usage: (function ARG)  */)
-  (Lisp_Object args)
-{
-  Lisp_Object quoted = XCAR (args);
-
-  if (!NILP (XCDR (args)))
-    xsignal2 (Qwrong_number_of_arguments, Qfunction, Flength (args));
-
-  if (!NILP (Vinternal_interpreter_environment)
-      && CONSP (quoted)
-      && EQ (XCAR (quoted), Qlambda))
-    { /* This is a lambda expression within a lexical environment;
-	 return an interpreted closure instead of a simple lambda.  */
```

Comment:
This is non-trivial code and I think these comments really help. Could we add them to the Rust too?

Reply:
Thank you. I had intended to restore them and failed to do so.